### PR TITLE
HOMS-95 Show message  when user not found in Activiti

### PR DESCRIPTION
**Pull request by GideonEXE**

### CODE REVIEW

File: hbw/app/controllers/hbw/users_controller.rb [hunk 1]
```diff
@@ -10,6 +10,10 @@ def lookup
       render :index
     end
 
+    def check_user
+      render json: {user_exist: widget.user_exist?(current_user_identifier)}.to_json
```
Comment (flash-gordon) [inline hbw/app/controllers/hbw/users_controller.rb]: `.to_json` is already called internally, just remove it
also, please, use spaces on both sides of curly brackets, it's the silent style guide of this project (see https://github.com/bbatsov/ruby-style-guide#spaces-braces for details)

File: hbw/app/models/hbw/activiti/adapter.rb [hunk 2]
```diff
@@ -15,6 +15,13 @@ def initialize(entity_code_key: , activiti_api: )
       def users
         HBW::BPMUser.fetch_all
       end
+      
+      def user_exist?(user_email)
+        user = HBW::BPMUser.with_connection(api) do
+          HBW::BPMUser.fetch(user_email)
+        end
+        user.present?
```
Comment (flash-gordon) [inline hbw/app/models/hbw/activiti/adapter.rb]: I'd prefer to use native ruby's `nil?` like this `!user.nil?`. AS' `present?` is an inversion of `blank?` and `blank?` does too much, and we don't like it :)

File: hbw/app/assets/javascripts/hbw/widget.js.jsx.coffee [hunk 3]
```diff
@@ -9,6 +9,12 @@ modulejs.define(
 
       constructor: (@options) ->
         connection = new Connection(path: @options.widgetPath, host: @options.widgetHost)
+        connection.request(
```
Comment (flash-gordon) [inline hbw/app/assets/javascripts/hbw/widget.js.jsx.coffee]: pls, move to a separate method, also place the call to the end of the constructor

**Comment by GideonEXE** [comment]
UPD

File: app/assets/javascripts/application.js.coffee [hunk 4]
```diff
@@ -203,4 +203,8 @@ $ ->
 
   widget.env.dispatcher.bind 'hbw:go-to-entity', 'host', (payload) ->
     window.location = payload.task.entity_url
+
+  widget.env.dispatcher.bind 'hbw:activiti_user_not_found', 'widget', () ->
```
Comment (flash-gordon) [inline app/assets/javascripts/application.js.coffee]: empty parens are still not required

File: hbw/app/assets/javascripts/hbw/widget.js.jsx.coffee [hunk 5]
```diff
@@ -77,4 +78,11 @@ modulejs.define(
 
       subscribeOnTasks: =>
         @tasksSubscription = @env.connection.subscribe(client: 'root', path: 'tasks')
+
+      checkActivitiUser: =>
+        @env.connection.request(
+          url: @env.connection.serverURL + '/users/check_user'
+          method: 'GET'
+          contentType: 'application/json').done((data) =>
+          @env.dispatcher.trigger('hbw:activiti_user_not_found') if !data.user_exist)
```
Comment (flash-gordon) [inline hbw/app/assets/javascripts/hbw/widget.js.jsx.coffee]: event names still prefer dashed names

File: hbw/app/assets/javascripts/hbw/widget.js.jsx.coffee [hunk 6]
```diff
@@ -77,4 +78,11 @@ modulejs.define(
 
       subscribeOnTasks: =>
         @tasksSubscription = @env.connection.subscribe(client: 'root', path: 'tasks')
+
+      checkActivitiUser: =>
+        @env.connection.request(
+          url: @env.connection.serverURL + '/users/check_user'
```
Comment (flash-gordon) [inline hbw/app/assets/javascripts/hbw/widget.js.jsx.coffee]: pls, remove to just `check`

**Comment by GideonEXE** [comment]
UPD

File: hbw/app/assets/javascripts/hbw/widget.js.jsx.coffee [hunk 7]
```diff
@@ -77,4 +78,11 @@ modulejs.define(
 
       subscribeOnTasks: =>
         @tasksSubscription = @env.connection.subscribe(client: 'root', path: 'tasks')
+
+      check: =>
+        @env.connection.request(
+          url: @env.connection.serverURL + '/users/check_user'
```
Comment (flash-gordon) [inline hbw/app/assets/javascripts/hbw/widget.js.jsx.coffee]: I meant controller action, not method name

**Comment by GideonEXE** [comment]
UPD

**Pull request merged**